### PR TITLE
Add dynamicFilterSplitsProcessed to OperatorStats

### PR DESCRIPTION
**Pull request by raunaqmorarka**
Exposing how many splits were processed with dynamic filter in table scan via OperatorContext and OperatorStats. (https://github.com/prestosql/presto/issues/3005)
E.g.:
```
{
      "stageId" : 1,
      "pipelineId" : 2,
      "operatorId" : 0,
      "planNodeId" : "355",
      "operatorType" : "ScanFilterAndProjectOperator",
      "totalDrivers" : 2002,
      "addInputCalls" : 662561,
      "addInputWall" : "11.32m",
      "addInputCpu" : "0.00ns",
      "physicalInputDataSize" : "48148296B",
      "physicalInputPositions" : 661815000,
      "internalNetworkInputDataSize" : "0B",
      "internalNetworkInputPositions" : 0,
      "rawInputDataSize" : "48148296B",
      "inputDataSize" : "3309072490B",
      "inputPositions" : 661815000,
      "sumSquaredInputPositions" : 1.023537125E15,
      "getOutputCalls" : 665179,
      "getOutputWall" : "21.65m",
      "getOutputCpu" : "24.25s",
      "outputDataSize" : "9265405160B",
      "outputPositions" : 661815000,
      "physicalWrittenDataSize" : "0B",
      "blockedWall" : "0.00ns",
      "finishCalls" : 0,
      "finishWall" : "0.00ns",
      "finishCpu" : "0.00ns",
      "userMemoryReservation" : "0B",
      "revocableMemoryReservation" : "0B",
      "systemMemoryReservation" : "0B",
      "peakUserMemoryReservation" : "0B",
      "peakSystemMemoryReservation" : "40278B",
      "peakRevocableMemoryReservation" : "0B",
      "peakTotalMemoryReservation" : "40278B",
      "spilledDataSize" : "0B",
      "dynamicFilterSplitsProcessed" : 2002
}
```

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 1]
```diff
@@ -557,6 +573,8 @@ public OperatorStats add(Iterable<OperatorStats> operators)
 
                 blockedReason,
 
+                dynamicFilterSplitsProcessed,
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]: move this above `physicalWrittenDataSize` to preserve ordering

File: presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java [hunk 2]
```diff
@@ -199,11 +199,14 @@ private void workProcessorOperatorStateMonitor(WorkProcessor.ProcessState<Page>
             long deltaInputDataSize = deltaAndSet(context.inputDataSize, sourceOperator.getInputDataSize().toBytes());
             long deltaInputPositions = deltaAndSet(context.inputPositions, sourceOperator.getInputPositions());
 
+            long deltaDynamicFilterSplitsProcessed = deltaAndSet(context.dynamicFilterSplitsProcessed, sourceOperator.getDynamicFilterSplitsProcessed());
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/WorkProcessorPipelineSourceOperator.java]: move this below `deltaReadTimeNanos`

File: presto-main/src/main/java/io/prestosql/operator/OperatorContext.java [hunk 3]
```diff
@@ -216,6 +218,11 @@ public void recordPhysicalWrittenData(long sizeInBytes)
         physicalWrittenDataSize.getAndAdd(sizeInBytes);
     }
 
+    public void recordDynamicFilterSplitProcessed(long dynamicFilterSplits)
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorContext.java]: move this above `recordPhysicalWrittenData`

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 4]
```diff
@@ -81,6 +81,8 @@
 
     private final Optional<BlockedReason> blockedReason;
 
+    private final long dynamicFilterSplitsProcessed;
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]: move this above `physicalWrittenDataSize`

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 5]
```diff
@@ -403,6 +409,12 @@ public DataSize getSpilledDataSize()
         return blockedReason;
     }
 
+    @JsonProperty
+    public long getDynamicFilterSplitsProcessed()
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]: move this above `getPhysicalWrittenDataSize`

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 6]
```diff
@@ -457,6 +469,8 @@ public OperatorStats add(Iterable<OperatorStats> operators)
 
         Optional<BlockedReason> blockedReason = this.blockedReason;
 
+        long dynamicFilterSplitsProcessed = this.dynamicFilterSplitsProcessed;
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]: move this above `long physicalWrittenDataSize = this.physicalWrittenDataSize.toBytes();`

File: presto-main/src/main/java/io/prestosql/operator/OperatorStats.java [hunk 7]
```diff
@@ -504,6 +518,8 @@ public OperatorStats add(Iterable<OperatorStats> operators)
                 blockedReason = operator.getBlockedReason();
             }
 
+            dynamicFilterSplitsProcessed += operator.getDynamicFilterSplitsProcessed();
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/OperatorStats.java]: move this above `physicalWrittenDataSize += operator.getPhysicalWrittenDataSize().toBytes();`

File: presto-main/src/main/java/io/prestosql/operator/TableScanOperator.java [hunk 8]
```diff
@@ -293,6 +293,9 @@ public Page getOutput()
             return null;
         }
         if (source == null) {
+            if (!dynamicFilter.get().isAll()) {
```
Comment (sopel39) [inline presto-main/src/main/java/io/prestosql/operator/TableScanOperator.java]: there is also `TableScanWorkProcessorOperator`

Comment (raunaqmorarka) [reply presto-main/src/main/java/io/prestosql/operator/TableScanOperator.java]: Made changes there as well

**Comment by sopel39** [comment]
merged, thanks!

**Pull request merged**